docs/reference/content/driver-reactive/tutorials/change-streams.md

```diff
@@ -14,98 +14,86 @@ MongoDB 3.6 introduces a new [`$changeStream`](http://dochub.mongodb.org/core/ch
-    @Override
-    public void apply(final ChangeStreamDocument<Document> changeStreamDocument) {
-        System.out.println(changeStreamDocument);
-    }
-};
-```
+{{% note class="important" %}}
+This guide uses the `Subscriber` implementations as covered in the [Quick Start Primer]({{< relref "driver-reactive/getting-started/quick-start-primer.md" >}}).
+{{% /note %}}
 
 ## Connect to a MongoDB Deployment
 
-Connect to a MongoDB deployment and declare and define a `MongoDatabase` and a `MongoCollection` instances.
+Connect to a MongoDB deployment and declare and define a `MongoDatabase` and a `MongoCollection` instance.
 
 For example, include the following code to connect to a replicaSet MongoDB deployment running on localhost on ports `27017`, `27018` and `27019`. 
 It also defines `database` to refer to the `test` database and `collection` to refer to the `restaurants` collection.
 
 ```java
-MongoClient mongoClient = MongoClients.create(new ConnectionString("mongodb://localhost:27017,localhost:27018,localhost:27019"));
+MongoClient mongoClient = MongoClients.create("mongodb://localhost:27017,localhost:27018,localhost:27019");
 MongoDatabase database = mongoClient.getDatabase("test");
 MongoCollection<Document> collection = database.getCollection("restaurants");
 ```
 
-For additional information on connecting to MongoDB, see [Connect to MongoDB]({{< relref "driver-async/tutorials/connect-to-mongodb.md" >}}).
+For additional information on connecting to MongoDB, see [Connect to MongoDB]({{< ref "connect-to-mongodb.md">}}).
 
 ## Watch the collection
 
-To create a change stream use one the [`MongoCollection.watch()`]({{<apiref "com/mongodb/async/client/MongoCollection.html#watch">}}) 
-methods.
+To create a change stream use one of the [`MongoCollection.watch()`]({{<apiref "com/mongodb/client/MongoCollection.html#watch">}}) methods.
 
 In the following example, the change stream prints out all changes it observes.
 
 ```java
-collection.watch().forEach(printBlock, callbackWhenFinished);
+OperationSubscriber<Document> subscriber = new PrintDocumentSubscriber();
+collection.watch().subscribe(subscriber);
+subscriber.await(); // Block on the subscriber
 ```
 
 ## Watch the database
 
 New in the 3.8 driver and MongoDB 4.0, applications can open a single change stream to watch all non-system collections of a database. To
-create such a change stream use one of the [`MongoDatabase.watch()`]({{<apiref "com/mongodb/async/client/MongoDatabase.html#watch">}}) 
-methods.
+create such a change stream use one of the [`MongoDatabase.watch()`]({{<apiref "com/mongodb/client/MongoDatabase.html#watch">}}) methods.
 
 In the following example, the change stream prints out all the changes it observes on the given database.
 
 ```java
-database.watch().forEach(printBlock, callbackWhenFinished);
+subscriber = new PrintDocumentSubscriber();
+database.watch().subscribe(subscriber);
+subscriber.await();
```

Comment:
This is one place where even in an example I wouldn't include an `await`, as it will generally never return.

Reply:
Well you have to block on something.  Not sure of an alternative without a busy loop.

Reply:
Not sure what you mean.  Why does the example have to block?  I understand that in the QuickTour we have to block sometimes so that we can run multiple examples in one go, but it seems out of place in the reference docs.

Reply:
Done - removed it throughout apart from GridFS and FLE examples which are more copy & paste